src/uu/echo/src/echo.rs

```diff
@@ -127,10 +132,9 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
     };
 
     match execute(no_newline, escaped, values) {
-        Ok(_) => 0,
+        Ok(_) => Ok(()),
         Err(f) => {
-            show_error!("{}", f);
-            1
+            return Err(USimpleError::new(1, format!("{}", f)));
```

Comment:
Please use `UIoError` instead of `USimpleError`. The documentation on the error module should explain how to use it.

Reply:
Is
```rust 
Err(f) => Err(UError::from(f.map_err_context(|| format!("cannot print!")))), 
```
ok here?

Reply:
```suggestion
            Err(UError::from(f.map_err_context(|| format!("cannot print!")))), 
```

Reply:
That's close! I would make `execute` return `()` and then use `map_err_context` on the `io::Result`:
```rust
execute(...).map_err_context(...)
```

Reply:
That's close! I would make `execute` return `()` and then use `map_err_context` on the `io::Result`:
```rust
execute(...).map_err_context(...)
```